marlowe-dashboard-client/src/Contract/View.purs

```diff
@@ -60,25 +67,53 @@ renderCurrentState state =
               ]
               [ span_ $ [ text "Balances" ] ]
           ]
-      , div [ classNames [ "px-4", "py-2", "bg-white" ] ]
-          [ span [] [ text $ "Step " <> show stepNumber ]
-          , span [] [ text "Completed" ]
-          ]
-      , div [ classNames [ "px-4", "py-2", "bg-white" ] ]
-          [ renderTasks (executionState ^. _namedActions)
+      , div [ classNames [ "px-4", "bg-white" ] ]
+          -- FIXME: zeplin has border color #dfdfdf, see if it makes sense to add that one to the pallete
+          --        or if this gray is fine
+          [ div [ classNames [ "py-2.5", "flex", "items-center", "border-b", "border-gray" ] ]
+              [ span
+                  [ classNames [ "text-xl", "font-semibold", "flex-grow" ] ]
+                  [ text $ "Step " <> show stepNumber ]
+              , span
+                  -- [ "flex", "items-center", "justify-center", "px-4", "py-3", "leading-none", "disabled:opacity-50", "disabled:cursor-not-allowed",  ]
+                  [ classNames [ "flex-grow", "rounded-3xl", "bg-gray", "py-2", "flex", "items-center" ] ]
+                  [ Icons.timer' [ "pl-3" ]
+                  , span [ classNames [ "text-xs", "flex-grow", "text-center", "font-semibold" ] ]
+                      [ text "1hr 2mins left" ]
+                  ]
+              ]
+          , div [ classNames [ "py-2" ] ]
+              [ renderTasks walletDetails state
+              ]
           ]
       ]
 
-renderTasks :: forall p. Array NamedAction -> HTML p Action
-renderTasks actions =
+renderTasks :: forall p. WalletDetails -> State -> HTML p Action
+renderTasks walletDetails state =
   let
+    executionState = state ^. _executionState
+
+    actions = executionState ^. _namedActions
+
+    contract = executionState ^. _contract
+
+    getRoleEntry = case _ of
+      (PK _) -> Nothing
+      (Role roleName) -> Just roleName
+
+    roles :: Array String
+    roles = Set.toUnfoldable $ Set.mapMaybe getRoleEntry (getParties contract)
+
     -- FIXME: We fake the namedActions for development until we fix the semantics
     actions' =
       [ MakeDeposit (Role "into account") (Role "by") (Token "" "") $ fromInt 1500
       ]
   in
     -- FIXME: need to group by role
-    div [] $ actions' <#> renderAction
+    div []
+      [ span_ [ text $ "Role (" <> walletDetails.nickname <> ")" ]
+      , div [] $ actions' <#> renderAction
+      ]
```

Comment:
Yeah, I think here we're just going to have to use the roleName. I don't see how we can map this to a wallet nickname... 🤔